src/NServiceBus.Core/Routing/UnicastRoute.cs

```diff
@@ -42,24 +44,18 @@ public UnicastRoute(string physicalAddress)
             this.physicalAddress = physicalAddress;
         }
 
-        IEnumerable<UnicastRoutingTarget> IUnicastRoute.Resolve(Func<Endpoint, IEnumerable<EndpointInstance>> instanceResolver)
+        async Task<IEnumerable<UnicastRoutingTarget>> IUnicastRoute.Resolve(Func<Endpoint, Task<IEnumerable<EndpointInstance>>> instanceResolver)
         {
             if (physicalAddress != null)
             {
-                yield return UnicastRoutingTarget.ToTransportAddress(physicalAddress);
+                return EnumerableEx.Single(UnicastRoutingTarget.ToTransportAddress(physicalAddress));
             }
-            else if (instance != null)
+            if (instance != null)
             {
-                yield return UnicastRoutingTarget.ToEndpointInstance(instance);
-            }
-            else
-            {
-                var instances = instanceResolver(endpoint);
-                foreach (var i in instances)
-                {
-                    yield return UnicastRoutingTarget.ToEndpointInstance(i);
-                }
+                return EnumerableEx.Single(UnicastRoutingTarget.ToEndpointInstance(instance));
             }
+            var instances = await instanceResolver(endpoint);
```

Comment:
`ConfigureAwait(false)`
Is there no way we can turn this into a return without await?